### PR TITLE
refactor: KeyMsg -> KeyPressMsg

### PR DESCRIPTION
**Pull request by bashbunni**
Cherry picked a single commit with changes from `alpha branch` with latest master

### CODE REVIEW

**Comment by bashbunni** [comment]
Looking into why the test is failing

**Comment by bashbunni** [comment]
Tests aren't passing on this branch - terminal color codes don't work. Need to fix that before it's ready :)

File: viewport/viewport.go [hunk 1]
```diff
@@ -331,18 +331,15 @@ func (m Model) updateAsModel(msg tea.Msg) (Model, tea.Cmd) {
 			}
 		}
 
-	case tea.MouseMsg:
-		if !m.MouseWheelEnabled || msg.Action != tea.MouseActionPress {
-			break
-		}
+	case tea.MouseWheelMsg:
 		switch msg.Button {
```
Comment (aymanbagabas) [inline viewport/viewport.go:335]: With https://github.com/charmbracelet/bubbletea/pull/1111, this is going to be `msg.Button()`

**Review by aymanbagabas — COMMENTED** [review]
LGTM

**Comment by bashbunni** [comment]
@aymanbagabas do you have any thoughts on the colour issue? It seems the v2-exp lipgloss branch loses support for ansi colour codes as you can see when you run the tests on this branch. It doesn't handle ints like it used to for lipgloss.Color

**Comment by bashbunni** [comment]
on Bubble Tea `v2-exp` it's now `KeyPressMsg.Code` not `KeyPressMsg.Runes`. Need to update that here before merging

**Comment by bashbunni** [comment]
swapping out `[]rune` with `rune` in bubbles is also needed based on bubble tea's `v2-exp` branch

**Comment by aymanbagabas** [comment]
fwiw, i've pushed these changes to `v2-exp`

**Comment by bashbunni** [comment]
@aymanbagabas kk sounds good! I'll close this issue then

**Pull request closed without merging**